### ISSUE TITLE
Run my own robot occur a problem

### ISSUE THREAD

**Issue by Zcyyy** [leggedrobotics/ocs2#60]
Hi,
    When I configured my robot using ocs2_legged_robot as an example, the following error occurred:
![image](https://user-images.githubusercontent.com/27263450/207200693-a474aed7-3c61-4ae4-bb83-e0697b570797.png)
How should I define my input to ensure that the MPC runs successfully?

**Comment by FaresKchaou** [comment]
Hey! 
I am having the same problem! have you found a solution for this problem ?

**Comment by Zcyyy** [comment]
> Hey! I am having the same problem! have you found a solution for this problem ?

Yes, the problem was resolved after I modified the config file. Please make sure that the degrees of freedom of your robot match those specified in the config file. Additionally, ensure that the names of the robot's end effectors and joints are the same. You can also modify the weights of the Q and R matrices. These are all factors that can potentially affect the calculations. Of course, there may be other factors that I haven't mentioned.

**Comment by FaresKchaou** [comment]
thank you for your quick reply! can you please give me more inputs or some help! i want to test it with a real franka panda robot before testing it on a mobile manipulator. i am a bit confused right now. i will be thankful for some input, i am a bit stuck now.

**Comment by FaresKchaou** [comment]
thank you for your quick reply!  i want to test ocs2 with a real franka panda robot before testing it on a mobile manipulator.  i will be thankful for some input or some help!  what i still didn't exactly get is: if the real robot is replacing the MRT  so that  i need to run the panda controller and just apply the mpc_policy (joint velocities) from the mpc node and then send the real states of the joints via mpc_observation topic (i think this is wrong because the MRT node is getting also the mpc_target topic)? i would be very thankful if you provide me some small tips!

**Comment by Zcyyy** [comment]
> thank you for your quick reply!  i want to test ocs2 with a real franka panda robot before testing it on a mobile manipulator.  i will be thankful for some input or some help!  what i still didn't exactly get is: if the real robot is replacing the MRT  so that  i need to run the panda controller and just apply the mpc_policy (joint velocities) from the mpc node and then send the real states of the joints via mpc_observation topic (i think this is wrong because the MRT node is getting also the mpc_target topic)? i would be very thankful if you provide me some small tips!

According to my understanding of ocs2, you cannot replace MRT. MRT is an important part to get the robot state observation. You should focus on the OC part.

**Comment by ppap36** [comment]
I am sorry to raise a question that has no bussienss with your problem.
SORRY
but have you ever met the problem mentioned in issue #87 
sorry to bother you

**Comment by tac-dog** [comment]
@FaresKchaou  I want to test ocs2 with a real franka panda robot too.I subscribe directly to the joint velocities of the mpc_policy to send to the real franka_panda.Is there anything wrong with this approach?thanks.
